src/cdk/testing/event-objects.ts

```diff
@@ -63,23 +71,31 @@ export function createTouchEvent(type: string, pageX = 0, pageY = 0) {
  * Dispatches a keydown event from an element.
  * @docs-private
  */
-export function createKeyboardEvent(type: string, keyCode: number, target?: Element, key?: string) {
+export function createKeyboardEvent(type: string, keyCode: number = 0, key: string = '',
+                                    target?: Element, modifiers: ModifierKeys = {}) {
   let event = document.createEvent('KeyboardEvent') as any;
   let originalPreventDefault = event.preventDefault;
 
   // Firefox does not support `initKeyboardEvent`, but supports `initKeyEvent`.
   if (event.initKeyEvent) {
-    event.initKeyEvent(type, true, true, window, 0, 0, 0, 0, 0, keyCode);
+    event.initKeyEvent(type, true, true, window, modifiers.control, modifiers.alt, modifiers.shift,
+        modifiers.meta, keyCode);
   } else {
-    event.initKeyboardEvent(type, true, true, window, 0, key, 0, '', false);
+    let modifiersStr = (modifiers.control ? 'Control ' : '' + modifiers.alt ? 'Alt ' : '' +
+        modifiers.shift ? 'Shift ' : '' + modifiers.meta ? 'Meta' : '').trim();
```

Comment:
We should add a link to the API description entry of `initKeyboardEvent` (so that we know why we need to construct this string).

Also this can be a `const` right?